[BUG] Incorrect deserialize/serialize _geo_distance sorting field

### What is the bug?
`SearchRequest._DESERIALIZER` incorrectly deserialize `"validation_method": "STRICT"` inside `_geo_distance` sorting parameters.
Actually, this field is undocumented. So, should it just throw an error?
https://opensearch.org/docs/latest/search-plugins/searching-data/sort/#sorting-by-geo-distance

### How can one reproduce the bug?
Following code will incorrectly deserialize/serialize geo sorting query.
```
String geoSearchSortStr = "{\n" +
                    "  \"query\": {\n" +
                    "    \"match_all\": {}\n" +
                    "  },\n" +
                    "  \"sort\": [\n" +
                    "    {\n" +
                    "      \"_geo_distance\": {\n" +
                    "        \"unit\": \"mi\",\n" +
                    "        \"distance_type\": \"arc\",\n" +
                    "        \"coordinates\": [\n" +
                    "          {\n" +
                    "            \"lon\": -100.00,\n" +
                    "            \"lat\": 30.00\n" +
                    "          }\n" +
                    "        ],\n" +
                    "        \"validation_method\": \"STRICT\",\n" +
                    "        \"ignore_unmapped\": false,\n" +
                    "        \"order\": \"asc\"\n" +
                    "      }\n" +
                    "    }\n" +
                    "  ]\n" +
                    "}";
SearchRequest geoSearchRequest = SearchRequest._DESERIALIZER
        .deserialize(new JacksonJsonpParser(new JsonFactory().createParser(geoSearchSortStr)), new JacksonJsonpMapper());
System.out.println(geoSearchRequest.toJsonString());
```

This will print an incorrect serialized result. Note that, `validation_method` is a list of mixed of object and string.
```
{
  "query": {
    "match_all": {}
  },
  "sort": [
    {
      "_geo_distance": {
        "validation_method": [  <------
          {
            "lat": 30,
            "lon": -100
          },
          "STRICT"
        ],
        "distance_type": "arc",
        "ignore_unmapped": false,
        "order": "asc",
        "unit": "mi"
      }
    }
  ]
}
```


### What is the expected behavior?
It should either raise deserialization error when `validation_method` present in `_geo_distance` sorting parameters or support `validation_method` in sorting correctly.

### What is your host/environment?
opensearch-java 2.21.0

@Xtansia Is it possible to have immediate fix for this? Now, it's impossible to pass this such request to the server.
For example, when calling `search` function like this.
```
SearchResponse geoSearchResponse = client.search(geoSearchRequest, HashMap.class);
```
It will immediately raise this error.
```
Exception in thread "main" org.opensearch.client.opensearch._types.OpenSearchException: Request failed: [parse_exception] unsupported symbol [S] in geohash [STRICT]
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.extractAndWrapCause(ApacheHttpClient5Transport.java:1153)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.performRequest(ApacheHttpClient5Transport.java:158)
	at org.opensearch.client.opensearch.OpenSearchClient.search(OpenSearchClient.java:1386)
	at com.intuit.ips.example.OpenSearchClientExample.main(OpenSearchClientExample.java:252)
Caused by: org.opensearch.client.opensearch._types.OpenSearchException: Request failed: [parse_exception] unsupported symbol [S] in geohash [STRICT]
	at org.opensearch.client.transport.Endpoint.exceptionConverter(Endpoint.java:102)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.prepareResponse(ApacheHttpClient5Transport.java:547)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport.lambda$performRequestAsync$0(ApacheHttpClient5Transport.java:191)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport$1.completed(ApacheHttpClient5Transport.java:236)
	at org.opensearch.client.transport.httpclient5.ApacheHttpClient5Transport$1.completed(ApacheHttpClient5Transport.java:225)
	at org.apache.hc.core5.concurrent.BasicFuture.completed(BasicFuture.java:148)
	at org.apache.hc.core5.concurrent.ComplexFuture.completed(ComplexFuture.java:72)
	at org.apache.hc.client5.http.impl.async.InternalAbstractHttpAsyncClient$2$1.completed(InternalAbstractHttpAsyncClient.java:304)
	at org.apache.hc.core5.http.nio.support.AbstractAsyncResponseConsumer$1.completed(AbstractAsyncResponseConsumer.java:101)
	at org.apache.hc.core5.http.nio.entity.AbstractBinAsyncEntityConsumer.completed(AbstractBinAsyncEntityConsumer.java:87)
	at org.apache.hc.core5.http.nio.entity.AbstractBinDataConsumer.streamEnd(AbstractBinDataConsumer.java:83)
	at org.apache.hc.core5.http.nio.support.AbstractAsyncResponseConsumer.streamEnd(AbstractAsyncResponseConsumer.java:142)
	at org.apache.hc.client5.http.impl.async.HttpAsyncMainClientExec$1.streamEnd(HttpAsyncMainClientExec.java:251)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamHandler.dataEnd(ClientHttp1StreamHandler.java:279)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1StreamDuplexer.dataEnd(ClientHttp1StreamDuplexer.java:366)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1StreamDuplexer.onInput(AbstractHttp1StreamDuplexer.java:338)
	at org.apache.hc.core5.http.impl.nio.AbstractHttp1IOEventHandler.inputReady(AbstractHttp1IOEventHandler.java:64)
	at org.apache.hc.core5.http.impl.nio.ClientHttp1IOEventHandler.inputReady(ClientHttp1IOEventHandler.java:41)
	at org.apache.hc.core5.reactor.InternalDataChannel.onIOEvent(InternalDataChannel.java:143)
	at org.apache.hc.core5.reactor.InternalChannel.handleIOEvent(InternalChannel.java:51)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.processEvents(SingleCoreIOReactor.java:176)
	at org.apache.hc.core5.reactor.SingleCoreIOReactor.doExecute(SingleCoreIOReactor.java:125)
	at org.apache.hc.core5.reactor.AbstractSingleCoreIOReactor.execute(AbstractSingleCoreIOReactor.java:92)
	at org.apache.hc.core5.reactor.IOReactorWorker.run(IOReactorWorker.java:44)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -47,6 +47,7 @@ This section is for maintaining a changelog for all breaking changes for the cli
 - Added support for the Search Pipeline APIs ([#1442](https://github.com/opensearch-project/opensearch-java/pull/1442))
 - Added support for `simple_pattern` and `simple_pattern_split` tokenizers ([#1448](https://github.com/opensearch-project/opensearch-java/pull/1448), [#1451](https://github.com/opensearch-project/opensearch-java/pull/1451))
 - Added support for the Search response `phase_took` field ([#1449](https://github.com/opensearch-project/opensearch-java/pull/1449)) 
+- Added support for `GeoDistanceSort`'s `nested` and `validation_method` fields ([#1470](https://github.com/opensearch-project/opensearch-java/pull/1470))
 
 ### Dependencies
 

--- a/java-client/src/main/java/org/opensearch/client/opensearch/_types/GeoDistanceSort.java
+++ b/java-client/src/main/java/org/opensearch/client/opensearch/_types/GeoDistanceSort.java
@@ -35,6 +35,7 @@ package org.opensearch.client.opensearch._types;
 import jakarta.json.stream.JsonGenerator;
 import java.util.List;
 import java.util.function.Function;
+import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import org.opensearch.client.json.JsonpDeserializable;
 import org.opensearch.client.json.JsonpDeserializer;
@@ -42,6 +43,7 @@ import org.opensearch.client.json.JsonpMapper;
 import org.opensearch.client.json.JsonpSerializable;
 import org.opensearch.client.json.ObjectBuilderDeserializer;
 import org.opensearch.client.json.ObjectDeserializer;
+import org.opensearch.client.opensearch._types.query_dsl.GeoValidationMethod;
 import org.opensearch.client.util.ApiTypeHelper;
 import org.opensearch.client.util.ObjectBuilder;
 import org.opensearch.client.util.ObjectBuilderBase;
@@ -69,6 +71,12 @@ public class GeoDistanceSort implements SortOptionsVariant, JsonpSerializable {
     @Nullable
     private final DistanceUnit unit;
 
+    @Nullable
+    private final GeoValidationMethod validationMethod;
+
+    @Nullable
+    private final NestedSortValue nested;
+
     // ---------------------------------------------------------------------------------------------
 
     private GeoDistanceSort(Builder builder) {
@@ -81,7 +89,8 @@ public class GeoDistanceSort implements SortOptionsVariant, JsonpSerializable {
         this.ignoreUnmapped = builder.ignoreUnmapped;
         this.order = builder.order;
         this.unit = builder.unit;
-
+        this.validationMethod = builder.validationMethod;
+        this.nested = builder.nested;
     }
 
     public static GeoDistanceSort of(Function<Builder, ObjectBuilder<GeoDistanceSort>> fn) {
@@ -151,6 +160,22 @@ public class GeoDistanceSort implements SortOptionsVariant, JsonpSerializable {
     }
 
     /**
+     * API name: {@code validation_method}
+     */
+    @Nullable
+    public final GeoValidationMethod validationMethod() {
+        return this.validationMethod;
+    }
+
+    /**
+     * API name: {@code nested}
+     */
+    @Nullable
+    public final NestedSortValue nested() {
+        return this.nested;
+    }
+
+    /**
      * Serialize this object to JSON.
      */
     public void serialize(JsonGenerator generator, JsonpMapper mapper) {
@@ -189,7 +214,14 @@ public class GeoDistanceSort implements SortOptionsVariant, JsonpSerializable {
             generator.writeKey("unit");
             this.unit.serialize(generator, mapper);
         }
-
+        if (this.validationMethod != null) {
+            generator.writeKey("validation_method");
+            this.validationMethod.serialize(generator, mapper);
+        }
+        if (this.nested != null) {
+            generator.writeKey("nested");
+            this.nested.serialize(generator, mapper);
+        }
     }
 
     // ---------------------------------------------------------------------------------------------
@@ -249,6 +281,12 @@ public class GeoDistanceSort implements SortOptionsVariant, JsonpSerializable {
         @Nullable
         private DistanceUnit unit;
 
+        @Nullable
+        private GeoValidationMethod validationMethod;
+
+        @Nullable
+        private NestedSortValue nested;
+
         /**
          * API name: {@code mode}
          */
@@ -290,6 +328,29 @@ public class GeoDistanceSort implements SortOptionsVariant, JsonpSerializable {
         }
 
         /**
+         * API name: {@code validation_method}
+         */
+        public final Builder validationMethod(@Nullable GeoValidationMethod value) {
+            this.validationMethod = value;
+            return this;
+        }
+
+        /**
+         * API name: {@code nested}
+         */
+        public final Builder nested(@Nullable NestedSortValue value) {
+            this.nested = value;
+            return this;
+        }
+
+        /**
+         * API name: {@code nested}
+         */
+        public final Builder nested(@Nonnull Function<NestedSortValue.Builder, ObjectBuilder<NestedSortValue>> fn) {
+            return nested(fn.apply(new NestedSortValue.Builder()).build());
+        }
+
+        /**
          * Builds a {@link GeoDistanceSort}.
          *
          * @throws NullPointerException
@@ -319,6 +380,8 @@ public class GeoDistanceSort implements SortOptionsVariant, JsonpSerializable {
         op.add(Builder::ignoreUnmapped, JsonpDeserializer.booleanDeserializer(), "ignore_unmapped");
         op.add(Builder::order, SortOrder._DESERIALIZER, "order");
         op.add(Builder::unit, DistanceUnit._DESERIALIZER, "unit");
+        op.add(Builder::validationMethod, GeoValidationMethod._DESERIALIZER, "validation_method");
+        op.add(Builder::nested, NestedSortValue._DESERIALIZER, "nested");
 
         op.setUnknownFieldHandler((builder, name, parser, mapper) -> {
             builder.field(name);

